File: frontend/src/wire/comment.ts
export type Comment = {
  id: string;
  postId: string;
  authorUid: string;
  body: string;
  createdAt: number;
};

export type CommentDto = {
  id: string;
  post_id: string;
  author_uid: string;
  body: string;
  created_at: number;
};

export const toComment = (w: CommentDto): Comment => ({
  id: w.id,
  postId: w.post_id,
  authorUid: w.author_uid,
  body: w.body,
  createdAt: w.created_at,
});

File: posts/models/comment.go
package models

import "github.com/google/uuid"

type Comment struct {
	ID        uuid.UUID `db:"id" json:"id"`
	PostID    uuid.UUID `db:"post_id" json:"post_id"`
	AuthorUID uuid.UUID `db:"author_uid" json:"author_uid"`
	Body      string    `db:"body" json:"body"`
	CreatedAt int64     `db:"created_at" json:"created_at"`
}
